front/app/modules/commercial/analytics/admin/utils/resolution.ts

```diff
@@ -1,26 +1,29 @@
 // i18n
 import messages from '../messages';
+import { isMonth, isWeek } from './timeSeries';
 
 // typings
 import { IResolution } from 'components/admin/ResolutionControl';
 import { FormatMessage } from 'typings';
+import { DateRow, DateColumn } from '../typings';
 
 export const resolutionDeducer =
```

Comment:
since you are removing the type parameter on the function calls what is the default type? and what is the point of having a type argument here?

Reply:
There is no default type- the prefix that you pass will be used to determine the structure of the row that can be passed to the returned function. E.g.

```
const deduceResolution = resolutionDeducer('dimension_date_registration');
deduceResolution([{ 'dimension_date_registration.month': '2022-11' }]) // is allowed
deduceResolution([{ 'dimension_date_created': '2022-11' }]) // throw type error
```

Reply:
but isn't this function `<Prefix extends string>(prefix: Prefix) =>` expect to be call with a type parameter?

Reply:
It's optional, if you don't provide the type parameter typescript will infer it based on use. For example

```
const identity1 = (input) => input
const identity2 = <T,>(input: T): T => input;

identity1(100) // function has signature (input: any) => any
identity2(100) // function has signature (input: number) => number
```

Reply:
ah ok, thanks